packages/@n8n/task-runner-python/tests/integration/test_port.py

```diff
@@ -0,0 +1,11 @@
+import socket
+
+
+def find_free_port() -> int:
+    """Find and return a free port on localhost."""
+    
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Comment:
is there a risk that a parallel test or something could steal the port before we take it into use at the test, since we close the socket after this `with`?

Probably not a real issue in practice though

Reply:
Thanks for bringing this up! We don't want to introduce flakiness, no matter how unlikely. Please see [`6812a6a`](https://github.com/n8n-io/n8n/pull/20825/commits/6812a6a22b841540c51f3ad15890457e37895370)